evalml/model_understanding/graphs.py

```diff
@@ -690,15 +691,51 @@ def partial_dependence(
 
     _raise_value_error_if_mostly_one_value(feature_list, percentiles[1])
     wrapped = evalml.pipelines.components.utils.scikit_learn_wrapped_estimator(pipeline)
+
     try:
-        preds = sk_partial_dependence(
-            wrapped,
-            X=X,
-            features=features,
-            percentiles=percentiles,
-            grid_resolution=grid_resolution,
-            kind=kind,
-        )
+        if any(is_datetime):
+            timestamps = np.array(
+                [X_dt - pd.Timestamp("1970-01-01")] // np.timedelta64(1, "s")
+            ).reshape(-1, 1)
+            grid, values = _grid_from_X(
+                timestamps, percentiles=percentiles, grid_resolution=grid_resolution
+            )
+            grid_dates = pd.to_datetime(
+                pd.Series(grid.squeeze()), unit="s"
+            ).values.reshape(-1, 1)
```

Comment:
I like this, I think it's clever.  Converting to seconds after Unix time, yes?  So ultimately your using a large integer number to compute the grid.  Perhaps we can use this calculation to replace the dates with seconds after unix time, run the data through sklearn's stock partial dependence, then convert the datetime feature back after the fact?  To me, I think that would mitigate some of the code duplication and use of private funcs issues.